### PR TITLE
fixes #1601: Term excerpt performance

### LINKED ISSUE #1601: Hit Excerpts are slowing down queries too much even when not requested
The hit excerpts are causing a significant delay in processing phrases even when they are not being requested.  We believe this is because we are going through the trouble of adding phrase excerpts even when they are not needed during the phrase/within function evaluation.

### PR DESCRIPTION
**Pull request by ivakegg**
* Updated to avoid gathering phrase indexes/offsets when excerpts are not being requested.

### CODE REVIEW

**Review by lbschanno — DISMISSED** [review]
LGTM.

File: warehouse/query-core/src/main/java/datawave/query/iterator/QueryIterator.java [hunk 1]
```diff
@@ -1096,20 +1096,47 @@ private Range getDocumentRange(NestedQueryIterator<Key> documentSource) {
...
+                if (null == jexlEvaluationFunction) {
+                    jexlEvaluationFunction = new JexlEvaluation(query, arithmetic);
+                }
             }
         }
+        
+        // update the jexl evaluation to gather phrase offsets if required for excerpts
+        if (getExcerptFields() != null && !getExcerptFields().isEmpty()) {
+            jexlEvaluationFunction.setGatherPhraseOffsets(true);
+        }
```
Comment (apmoriarty) [inline warehouse/query-core/src/main/java/datawave/query/iterator/QueryIterator.java:1138]: Might want to create an issue to diff the excerpt fields against the fields in the query's content function. It's possibly (though unlikely) that the content function fields in the query do not match the fields specified in the excerpt fields -- as the result of query model expansion and/or range stream pruning.

**Review by apmoriarty — DISMISSED** [review]
LGTM -- commented for additional follow on work.

File: warehouse/query-core/src/main/java/datawave/query/iterator/QueryIterator.java (hunk 1 above)
Comment (ivakegg) [reply warehouse/query-core/src/main/java/datawave/query/iterator/QueryIterator.java:1138]: #1607

**Pull request merged**